analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.HasConstraint.cs

```diff
@@ -0,0 +1,118 @@
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn;
+
+public partial class ProgramStateTest
+{
+    private static IEnumerable<object[]> HasConstraintTestData => new[]
+    {
+        new object[] { null, ObjectConstraint.Null, false },
+        new object[] { SymbolicValue.Empty, ObjectConstraint.Null, false },
+        new object[] { SymbolicValue.Null, ObjectConstraint.Null, true },
+        new object[] { SymbolicValue.Null, ObjectConstraint.NotNull, false },
+        new object[] { SymbolicValue.Null, BoolConstraint.True, false }
+    };
+
+    [TestMethod]
+    [DynamicData(nameof(HasConstraintTestData))]
+    public void HasConstraint_Operation(SymbolicValue symbolicValue, SymbolicConstraint constraint, bool hasConstraint)
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("var s = \"\";");
+        var operation = cfg.Blocks[1].Operations[0];
+        var state = ProgramState.Empty;
+        if (symbolicValue is not null)
+        {
+            state = state.SetOperationValue(operation, symbolicValue);
+        }
+        state.HasConstraint(operation, constraint).Should().Be(hasConstraint);
+    }
+
+    [TestMethod]
+    public void HasConstraint_Operation_IsNullSafe() =>
+        ProgramState.Empty.HasConstraint((IOperation)null, ObjectConstraint.Null).Should().BeFalse();
+
+    [TestMethod]
+    [DynamicData(nameof(HasConstraintTestData))]
+    public void HasConstraint_Symbol(SymbolicValue symbolicValue, SymbolicConstraint constraint, bool hasConstraint)
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("var s = \"\";");
+        var symbol = cfg.Blocks[1].Operations[0].ChildOperations.First().TrackedSymbol();
+        var state = ProgramState.Empty;
+        if (symbolicValue is not null)
+        {
+            state = state.SetSymbolValue(symbol, symbolicValue);
+        }
+        state.HasConstraint(symbol, constraint).Should().Be(hasConstraint);
+    }
+
+    [TestMethod]
+    public void HasConstraint_Symbol_IsNullSafe() =>
+        ProgramState.Empty.HasConstraint((ISymbol)null, ObjectConstraint.Null).Should().BeFalse();
+
+    private static IEnumerable<object[]> ConstraintTestData => new[]
+    {
+        new object[] { null, null },
+        new object[] { SymbolicValue.Empty, null },
+        new object[] { SymbolicValue.Null, null },
+        new object[] { SymbolicValue.True, BoolConstraint.True },
+        new object[] { SymbolicValue.False, BoolConstraint.False }
+    };
+
+    [TestMethod]
+    [DynamicData(nameof(ConstraintTestData))]
+    public void Constraint_Operation(SymbolicValue symbolicValue, SymbolicConstraint constraint)
```

Comment:
The filename is `.HasConstraint.cs` but it also contains `Constraint` methods. What about `.Constrain.cs` instead to cover both?